src/main/java/core/basesyntax/WorkWithFile.java

```diff
@@ -1,7 +1,69 @@
 package core.basesyntax;
 
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+
 public class WorkWithFile {
+    private static final String NAME_OF_OPERATION_SUPPLY = "supply";
+    private static final String NAME_OF_OPERATION_BUY = "buy";
+    private static final int INDEX_OF_OPERATION_TYPE = 0;
+    private static final int INDEX_OF_AMOUNT = 1;
+    private static final int INDEX_OF_SUPPLY_AMOUNT = 0;
+    private static final int INDEX_OF_BUY_AMOUNT = 1;
+
     public void getStatistic(String fromFileName, String toFileName) {
+        int[] operationsAmounts = readAmountsOfOperations(fromFileName);
+        int supplyAmount = operationsAmounts[INDEX_OF_SUPPLY_AMOUNT];
+        int buyAmount = operationsAmounts[INDEX_OF_BUY_AMOUNT];
+
+        String report = createReportMessage(supplyAmount, buyAmount);
+        writeReportToFile(report, toFileName);
+    }
+
+    private static int[] readAmountsOfOperations(String fromFileName) {
+        int supplyAmount = 0;
+        int buyAmount = 0;
+
+        try (BufferedReader reader = new BufferedReader(new FileReader(fromFileName))) {
+            String value = reader.readLine();
+            String[] lineFromInputfile;
+            while (value != null) {
+                lineFromInputfile = value.split(",");
+                String operationType = lineFromInputfile[INDEX_OF_OPERATION_TYPE];
+                int amount = Integer.parseInt(lineFromInputfile[INDEX_OF_AMOUNT]);
+                if (operationType.equals(NAME_OF_OPERATION_SUPPLY)) {
+                    supplyAmount += amount;
+                } else if (operationType.equals(NAME_OF_OPERATION_BUY)) {
+                    buyAmount += amount;
+                }
+                value = reader.readLine();
+            }
+        } catch (IOException e) {
+            throw new RuntimeException("Can't read file", e);
+        }
+        return new int[]{supplyAmount, buyAmount};
+    }
+
+    private static String createReportMessage(int supplyAmount, int buyAmount) {
+        final int resultAmount = supplyAmount - buyAmount;
+        StringBuilder stringBuilderResult = new StringBuilder();
+        stringBuilderResult.append(NAME_OF_OPERATION_SUPPLY).append(",")
```

Comment:
create constants for result and comma symbol

Reply:
Ok, if with coma symbol it's clear, then with constant for result. It will be like there:
* private static int resultAmount; 
but not final, because I have to calculate this while the program is running. Or you mean something else?

Reply:
I've watched the coments below another works and finnaly got it :)
You was talking about "result" in StrindBuilder)